### PR TITLE
sshd: update SSH port to 2200

### PR DESCRIPTION
**Pull request by svenrademakers**
default ssh config changes:
* enabled max authentication retries to 3
* disabled TCP forwarding
* disallowed emptyPasswords

### CODE REVIEW

File: tp2bmc/board/tp2bmc/overlay/etc/ssh/sshd_config [hunk 1]
```diff
@@ -82,7 +82,7 @@ AuthorizedKeysFile	.ssh/authorized_keys
 #UsePAM no
 
 #AllowAgentForwarding yes
-#AllowTcpForwarding yes
+AllowTcpForwarding no
```
Comment (ruslashev) [inline tp2bmc/board/tp2bmc/overlay/etc/ssh/sshd_config:85]: Why was this disabled? It doesn't really improve security, per sshd_config(5). Perhaps a setting of `local` would suffice?

**Comment by ruslashev** [comment]
> Is there a solid rationale for a nonstandard port?

I was the initiator for this change, and it was indeed to provide a small, but not insignificant obstacle to automated scanners. I just think it's embarrassing that we expose a short, easily-guessable, single-word, publicly-available password on a default port :22, which should be hammered by rouge worldwide scanners hourly.

> If we're looking to harden sshd, firewall rules and stricter authentication policies (like the other changes here) are better ways of achieving solid security.

Indeed, I would like to eventually require users to provide authorized_keys for pubkey auth, or (thanks for the reminder) provide firewall rules, but I don't think we should retain a combo of this password + this port + PermitRootLogin.

I am aware that sophisticated scanners could just add a rule sourced from this repository, but at least this change would impact those who don't. Do you agree?

**Comment by CFSworks** [comment]
Strengthening the default password would probably be the tack I'd take: `turingpi2` offers better (but still not great) resistance against dictionary attacks. We can do more if we're willing to get creative -- plus, a harder-to-remember default password has the side bonus of encouraging the user to change it. :)

If we really insist on changing the SSH port, `2200` is not a great alternate. `220`, `2200`, `2222`, and so on are already popular enough that (more sophisticated) scanners do check for them. Arguably, it's worse if the first automated attack to succeed comes from one of the more sophisticated adversaries. (But "no successful attack" is still the best option.)

I'd still prefer something comparable to fail2ban for this kind of thing, however. Perhaps a login nag when the user hasn't changed the default password, too?

Given the state of web auth, though, we really should be telling users to keep their BMCs inaccessible from the open Internet for now.

**Comment by ruslashev** [comment]
`turingpi2` works for me, plus I initially suggested 2202 to Sven, although it's the same marginal improvement. I also want to add a large disclaimer in the docs urging users to change their default password.

> fail2ban

Agree, and the good news is that it's readily available in the buildroot packages.

> Perhaps a login nag when the user hasn't changed the default password, too?

I thought about this previously, but was unsure of how to perform this, and perform securely. It should be possible to change post-build script to get a hashed password from /etc/shadow and use it in `~/.profile`, but it seemed to me like an approach not worth implementing.

> web auth

That's also upcoming, thankfully.

**Comment by svenrademakers** [comment]
We all agree this is a poor man's solution. @ruslashev I wrote some documentation on key-based authentication, which I want to advise users to set up if they expose ssh to the outside. Disabling the root login is also going to be part of this. 

Firewall rules is a very good idea, but is this in scope of TP? as its very network dependant.

Maybe this PR was a bit too spontaneous, but I'm happy there is a discussion going on, with some good suggestions :)

We should think about how can we serve/protect entry-level users best? Advanced users will edit sshd to their liking anyway.

**Comment by svenrademakers** [comment]
Closing PR for now. Proposed change is not significant nor contributing to a more secure target.

**Pull request closed without merging**